demo/docker/Dockerfile

```diff
@@ -1,22 +1,20 @@
-FROM python:3.6
-
-ENV USER=root
+FROM python:3.7-slim-buster AS base
 
 # Upgrade or install packages needed
 RUN apt-get update && apt-get -y install \
     g++ \
-    gcc \
-    binutils \
-    clang \ 
     cmake 
 
-# Install xlearn
-RUN git clone https://github.com/aksnzhy/xlearn.git 
-RUN cd xlearn && ./build.sh
-# Install liblinear
-RUN cd ..
-RUN git clone https://github.com/cjlin1/liblinear.git
-RUN cd liblinear/python/ && make
+# Install xlearn and liblinear
+RUN pip install xlearn==0.40a1 liblinear==2.11.2
+
+FROM python:3.7-slim-buster AS lib
+COPY --from=base /usr/local/lib/python3.7/site-packages /usr/local/lib/python3.7/
+
+ENV USER=root
```

Comment:
@ivotron When I remove the root line I'm not able to run the tests from the demo, do you know why this happens?

Reply:
not sure. can you paste the commands you're using to run the demo so I can reproduce the issue on my end?

Reply:
Sure. So once I run the image this is what I do:
- cd xlearn/demo/classification/higgs/
- python run_higgs.py

Reply:
do you that inside a container? or as part of building the dockerfile? I'll ask on slack

Reply:
it looks like the library assumes the `USER` variable exists, otherwise fails: https://github.com/aksnzhy/xlearn/blob/43fd6819d5b514099db7b85b27ebd010e0ea230e/src/base/system.h#L48 . This is a bit unorthodox but that's what it does, so we need to define it as you had it.

Also, I hadn't realized that the package on pypi is quite old. So we will have to build from source as you were doing before, sorry! This is what it ended up working on my end:

```dockerfile
ROM python:3.7-slim-buster as base

RUN apt update && \
    apt install -y cmake g++ && \
    rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*

COPY . /xlearn

# build from source
RUN cd /xlearn && \
    ls -l && \
    ./build.sh && \
    rm -r /xlearn

# install other library
RUN pip install --no-cache-dir liblinear==2.11.2

# create an image without build dependencies
FROM python:3.7-slim-buster AS lib
ENV USER=root
COPY --from=base /usr/local/lib/python3.7/site-packages/* /usr/local/lib/python3.7/
```

I built the above with:

```bash
cd xlearn/
docker build -t xlearn -f demo/docker/Dockerfile .
```

And then ran it with:

```bash
docker run --rm \
  --volume $PWD:/workspace \
  --workdir=/ws/demo/classification/higgs/ \
  xlearn \
    python run_higgs.py
```

Reply:
Awesome, thanks!